lms/static/js/learner_dashboard/views/course_entitlement_view.js

```diff
@@ -0,0 +1,395 @@
+
+                     // Grab URLs that handle changing of enrollment and entering a newly enrolled session.
+                     this.enrollUrl = options.enrollUrl;
+                     this.courseHomeUrl = options.courseHomeUrl;
+
+                     // Grab elements from the parent card that work with this view and bind associated events
+                     this.$triggerOpenBtn = $(options.triggerOpenBtn); // Opens/closes session selection view
+                     this.$dateDisplayField = $(options.dateDisplayField); // Displays current session dates
+                     this.$enterCourseBtn = $(options.enterCourseBtn); // Button link to course home page
+                     this.$courseCardMessages = $(options.courseCardMessages); // Additional session messages
+                     this.$courseTitleLink = $(options.courseTitleLink); // Title link to course home page
+                     this.$courseImageLink = $(options.courseImageLink); // Image link to course home page
+                     this.$triggerOpenBtn.on('click', this.toggleSessionSelectionPanel.bind(this));
+
+                     this.render(options);
+                     this.postRender();
+                 },
+
+                 render: function() {
+                     HtmlUtils.setHtml(this.$el, this.tpl(this.entitlementModel.toJSON()));
+                     this.delegateEvents();
+                     this.updateEnrollBtn();
+                 },
+
+                 postRender: function() {
+                     // Close popover on click-away
+                     $(document).on('click', function(e) {
+                         if (!($(e.target).closest('.enroll-btn-initial, .popover').length)) {
+                             this.$('.enroll-btn-initial').popover('hide');
+                         }
+                     }.bind(this));
+
+                     // Ensure that focus moves to the popover on click of the initial change enrollment button.
+                     $(document).on('click', '.enroll-btn-initial', function() {
+                         this.showDialog($('.enroll-btn-initial'));
+                         this.$('.final-confirmation-btn:first').focus();
+                     }.bind(this));
+                 },
+
+                 handleEnrollChange: function() {
+                     /*
+                     Handles enrolling in a course, unenrolling in a session and changing session.
+                     The new session id is stored as a data attribute on the option in the session-select element.
+                     */
+                     var isUnenrolling;
+
+                     // Grab the id for the desired session, an unenrollment event will return null
+                     this.currentSessionSelection = this.$('.session-select')
+                         .find('option:selected').data('session_id');
+                     isUnenrolling = !this.currentSessionSelection;
```

Comment:
By unenroll, you mean 'leaving the session' right? not destroying the entire entitlement/enrollment?

Nit: The term Unenroll seems to be overloaded in our application, so if the meaning here is 'leave session', could we call this something like 'isLeavingSession'?

Reply:
Yes, good call, updated the session

Reply:
Great catch, @amangano-edx. @HarryRein I suspect "enroll" is already widespread even in the new code, but if it's helpful to lessen any potential confusion, we were trying to reserve the verbs "enroll/unenroll" for courses and entitlements. Anything to do with sessions, we want to avoid using "enroll/unenroll", so we are going with "select", "change," "leave."